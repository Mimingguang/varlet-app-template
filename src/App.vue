<script setup lang="ts">
const msg = ref('按钮')
</script>

<template>
  <var-button>Varlet</var-button>
</template>
