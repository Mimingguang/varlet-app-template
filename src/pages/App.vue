<template>
  <var-button type="primary">{{ $t('hello') }}</var-button>
</template>
