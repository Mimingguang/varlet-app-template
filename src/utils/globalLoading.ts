export function removeGlobalLoading() {
  document.body.removeChild(document.querySelector('#app-global-loading')!)
}
