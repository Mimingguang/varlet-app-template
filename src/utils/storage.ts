export interface Storage {
  set(key: string, value: any): void
  get(key: string): any
  remove(key: string): void
}

const createStorage = (storage: globalThis.Storage) => {
  return {
    set(key: string, value: unknown) {
      if (typeof value !== 'string') {
        value = JSON.stringify(value)
      }
      storage.setItem(key, value as string)
    },
    get(key: string): any {
      const data = storage.getItem(key) as string
      try {
        return JSON.parse(data)
      } catch (err) {
        return data
      }
    },
    remove(key: string): void {
      storage.removeItem(key)
    }
  }
}

export const sessionStorage = createStorage(globalThis.sessionStorage)
export const localStorage = createStorage(globalThis.localStorage)
