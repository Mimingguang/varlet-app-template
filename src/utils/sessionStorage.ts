import { type Storage } from "./types/storage.type";

const sessionstorage: Storage = {
  set(key: string, value: any): void {
    if (typeof value === "object") {
      value = JSON.stringify(value);
    }
    localStorage.setItem(key, value);
  },
  get(key: string): any {
    const data = localStorage.getItem(key) as string;
    try {
      return JSON.parse(data);
    } catch (err) {
      return data;
    }
  },
  remove(key: string): void {
    localStorage.removeItem(key);
  },
};

export default sessionstorage;
