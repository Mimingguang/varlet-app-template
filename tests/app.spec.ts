import App from '@/App.vue'
import { mount } from './utils'
import { expect, it } from 'vitest'

it('test App.vue mount', () => {
  const wrapper = mount(App)

  expect(wrapper.html()).toMatchSnapshot()
})
