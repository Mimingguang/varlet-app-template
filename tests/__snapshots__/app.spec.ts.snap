// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`test App.vue mount 1`] = `
"<button class=\\"var-button var--box var-button--normal var--inline-flex var-button--primary var-elevation--2\\" type=\\"button\\">
  <!--v-if-->
  <div class=\\"var-button__content\\">mock i18n text</div>
  <div class=\\"var-hover-overlay\\"></div>
</button>"
`;
